Eastern-Research-Group/mywaterway
# Bugfix/Unmounted updates

## Main Changes:
* Implemented a `useAbortSignal` hook to abort async requests on component unmount.
* Replaced a `p` tag with a `div` tag to eliminate the "Validate DOM nesting" error. It's already wrapped in a `header`, so semantics shouldn't suffer.
* Addressed two instances where error messages were being shown in unwanted places.

## Steps To Test:
1. With the browser console open, navigate to the _Protect_ tab of the _Community_ page after choosing any location
2. Confirm that the warning "validateDOMNesting(...): \<div\> cannot appear as a descendant of \<p\>" does not appear in the console
3. Switch to the _Identified Issues_ tab, then use the search input to change to a different location
4. Confirm that the "Waterbody information is temporarily unavailable" error box is not visible while the new data loads
5. Enter gibberish into the search input, so that the "Data is not available for this location" error box is displayed
6. Switch to the _State_ page, then back to the _Community_ page, and confirm that the error box is not still visible
7. To test for the unmounted updates checks, navigate around the application with the console open, and confirm that the "Can't perform a React state update on an unmounted component" warnings are, at least, reduced
